### PR TITLE
feat: add isolation level configuration to SQL storage

### PR DESCRIPTION
**Pull request by julpayne**
## What and why

- Introduced an `isolationLevel` field in the `sqlStorage` struct to allow configuration of the SQL isolation level.
- Implemented `GetIsolationLevel` method in `SQLDatabaseConfig` to handle various isolation level inputs, including validation and trimming of whitespace.
- Added comprehensive unit tests for `GetIsolationLevel` to ensure correct behavior across different scenarios, including valid, invalid, and edge case inputs.
- Updated the transaction handling in sqlStorage to include the isolation level in error logs when a transaction fails to begin.

Assisted-by: Cursor

## Type

- [x] feat
- [ ] fix
- [ ] docs
- [ ] refactor / chore
- [ ] test / ci

## Testing

- [ ] Tests added or updated
- [x] Tested manually

## Release Notes

* **New Features**
  * Added SQL transaction isolation level configuration support with automatic defaults based on database driver type.
  * Environment variable `DEBUG_SQL_ISOLATION_LEVEL` enables explicit isolation level control.

* **Improvements**
  * Enhanced transaction error messages to include isolation level information for improved troubleshooting.

### CODE REVIEW

**Review by ppadashe-psp — APPROVED** [review]

**Review by mariusdanciu — COMMENTED** [review]
Code lgtm, but I don't see why this configuration would be needed. Why would an admin need to configure this ? It seems to me that the isolation level is determined by the application design not customizable by an admin.

**Comment by julpayne** [comment]
> Code lgtm, but I don't see why this configuration would be needed. Why would an admin need to configure this ? It seems to me that the isolation level is determined by the application design not customizable by an admin.

@mariusdanciu we can go with a default value but I was not sure that the value would be the same between development sqlite and production postgres? I have been using the option to do some tests because when running the tests in parallel there are deadlocks when using the in-memory sqlite (to be expected)?

**Comment by mariusdanciu** [comment]
> > Code lgtm, but I don't see why this configuration would be needed. Why would an admin need to configure this ? It seems to me that the isolation level is determined by the application design not customizable by an admin.
> 
> @mariusdanciu we can go with a default value but I was not sure that the value would be the same between development sqlite and production postgres? I have been using the option to do some tests because when running the tests in parallel there are deadlocks when using the in-memory sqlite (to be expected)?

Not sure why SQLLite WAL mode would lead to deadlock, but exposing this in config, may confuse an admin. How would an admin know what to use ? If we are running in deadlocks is a code problem that we need to address not and admin config matter - IMO.

SQLLite and Postgres can potentially have different isolation levels set at code level.

**Comment by julpayne** [comment]
@mariusdanciu I changed this so that we can test different versions using an environment variable - personally I think that keeping this in the config as an undocumented field is more consistent but an environment variable works

**Review by mariusdanciu — APPROVED** [review]
lgtm

File: internal/eval_hub/storage/sql/transaction.go [hunk 1]
```diff
@@ -33,12 +33,12 @@ func (s *sqlStorage) withTransaction(name string, resourceID string, fn Transact
...
 			s.logger.Error("Failed to commit transaction", "name", fmt.Sprintf("commit transaction %s", name), "resource_id", resourceID, "error", txnErr.Error())
-			return serviceerrors.NewServiceError(messages.DatabaseOperationFailed, "Type", fmt.Sprintf("commit transaction %s", name), "ResourceId", resourceID, "Error", txnErr.Error())
+			return serviceerrors.NewServiceError(messages.DatabaseOperationFailed, "Type", fmt.Sprintf("commit transaction %s", name), "ResourceId", resourceID, "isolation_level", s.isolationLevel.String(), "Error", txnErr.Error())
 		}
 	} else {
 		if txnErr := txn.Rollback(); txnErr != nil {
 			s.logger.Error("Failed to rollback transaction", "name", fmt.Sprintf("rollback transaction %s", name), "resource_id", resourceID, "error", txnErr.Error())
-			return serviceerrors.NewServiceError(messages.DatabaseOperationFailed, "Type", fmt.Sprintf("rollback transaction %s", name), "ResourceId", resourceID, "Error", txnErr.Error())
+			return serviceerrors.NewServiceError(messages.DatabaseOperationFailed, "Type", fmt.Sprintf("rollback transaction %s", name), "ResourceId", resourceID, "isolation_level", s.isolationLevel.String(), "Error", txnErr.Error())
```
[the remark this answers is not in the document]
Comment (julpayne) [reply internal/eval_hub/storage/sql/transaction.go]: Fixed
